File: Ruby/DeepSpaceRuby/lib/Weapon.rb
#encoding:utf-8
require_relative 'WeaponToUI'

module Deepspace
  class Weapon
    attr_reader :type, :uses, :name

    def initialize(n,t,u)
      @name = n
      @type = t
      @uses = u
    end

    def self.newCopy(w)
      new(w.name, w.type, w.uses)
    end

    public
    def power
      return @type.power
    end

    def useIt
      if @uses > 0 then
        @uses -= 1
        return power
      else
        return 1.0
      end
    end

    def to_s
      return "Nombre del arma: #{@name}, Tipo: #{@type}, Uses: #{@uses}"
    end

    def getUIversion
      return WeaponToUI.new(self)
    end
  end
end

File: Java/DeepspaceJava/deepspace/src/deepspace/Weapon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author paula
 */
public class Weapon {
    private String name;
    private WeaponType type;
    private int uses;

    Weapon (String n, WeaponType t, int u){
        name = n;
        type = t;
        uses = u;
    }

    Weapon (Weapon w){
        name = w.getName();
        type = w.getType();
        uses = w.getUses();
    }

    public String getName (){
        return name;
    }

    public WeaponType getType (){
        return type;
    }

    public int getUses (){
        return uses;
    }

    public float power (){
        return type.getPower();
    }

    public float useIt (){
        if (uses > 0){
            uses--;
            return power();
        }
        else
            return 1.0f;
    }
    
    public WeaponToUI getUIversion (){
      return new WeaponToUI(this);
    }
    
    @Override
    public String toString(){
        return "Nombre: "+this.name+" \nTipo: "+this.getType()+"("+power()+") \nUsos: "+this.getUses()+"  ";
    }
}
